Parallel build of `mlkem` target on MacOS with gcc-11 sometimes fails with libc++ runtime error

On MacOS, I observe spurious build failures in the gcc11 nix shell, when using parallel `make`:

```
% nix develop .#ci_gcc11
% for i in {0..10}; do make clean && CFLAGS=-O1 make mlkem OPT=1 AUTO=1 -j3 || break; done
```

Sooner or later, this fails with the following:

```
  AR      test/build/libmlkem512.a
  AR      test/build/libmlkem768.a
  AR      test/build/libmlkem1024.a
libc++abi: terminating due to uncaught exception of type std::runtime_error: opening input file: No such file or directory
/nix/store/y73x2c86csvisyy4yzz89fwg4r93ss4j-post-link-sign-hook: line 4:  9370 Abort trap: 6           CODESIGN_ALLOCATE=/nix/store/4wddwjq1nb41isiwm5cxrawx127wwcpx-cctools-llvm-16.0.6-973.0.1/bin/codesign_allocate /nix/store/fqilqr85zsrlrlbygjn862c16m5aiqrx-sigtool-0.1.3/bin/codesign -f -s - "$linkerOutput"
collect2: error: ld returned 134 exit status
make: *** [mk/rules.mk:24: test/build/libmlkem512.a] Error 1
make: *** Waiting for unfinished jobs....
libc++abi: terminating due to uncaught exception of type std::runtime_error: stat of a.out failed: No such file or directory
/nix/store/y73x2c86csvisyy4yzz89fwg4r93ss4j-post-link-sign-hook: line 4:  9371 Abort trap: 6           CODESIGN_ALLOCATE=/nix/store/4wddwjq1nb41isiwm5cxrawx127wwcpx-cctools-llvm-16.0.6-973.0.1/bin/codesign_allocate /nix/store/fqilqr85zsrlrlbygjn862c16m5aiqrx-sigtool-0.1.3/bin/codesign -f -s - "$linkerOutput"
collect2: error: ld returned 134 exit status
make: *** [mk/rules.mk:24: test/build/libmlkem768.a] Error 1
  AR         Checked for duplicated symbols
```

I don't see this without `-j3`, nor with another compiler version.

--- a/mk/rules.mk
+++ b/mk/rules.mk
@@ -30,15 +30,15 @@ ifneq ($(findstring Darwin,$(HOST_PLATFORM)),) # if is on macOS
 	$(Q)echo "int main(void) {return 0;}" \
 		| $(CC) -x c - -L$(BUILD_DIR) $(_CFLAGS) \
 		 -all_load -Wl,-undefined,dynamic_lookup -l$(_LIB) \
-		 -Imlkem $(wildcard test/notrandombytes/*.c)
-	$(Q)rm -f a.out
+		 -Imlkem $(wildcard test/notrandombytes/*.c) -o $(@:%.a=%_tmp.a.out)
+	$(Q)rm -f $(@:%.a=%_tmp.a.out)
 else                                           # if not on macOS
 	$(Q)echo "int main(void) {return 0;}" \
 		| $(CC) -x c - -L$(BUILD_DIR) $(_CFLAGS) \
 		-Wl,--whole-archive,--unresolved-symbols=ignore-in-object-files -l$(_LIB) \
 		-Wl,--no-whole-archive \
-		-Imlkem $(wildcard test/notrandombytes/*.c)
-	$(Q)rm -f a.out
+		-Imlkem $(wildcard test/notrandombytes/*.c) -o $(@:%.a=%_tmp.a.out)
+	$(Q)rm -f $(@:%.a=%_tmp.a.out)
 endif
 	$(Q)echo "  AR         Checked for duplicated symbols"
 

